### PR TITLE
Add the httperror package

### PR DESCRIPTION
**Pull request by johnworth**
Moves the error handling logic out of the logging package, re-adds the Echo-related code, and provides a README with a couple of examples.

### CODE REVIEW

**Comment by ianmcorvidae** [comment]
So, I did some changes to how this works in data-usage-api, because I needed it to do a bit more. I'll dig that up and leave comments here, hopefully we can unify the two.

**Comment by johnworth** [comment]
Sounds good.

**Comment by ianmcorvidae** [comment]
actually I think https://github.com/cyverse-de/data-usage-api/commit/fb6dbd285f0d2be7b92faf79b046e0c0663d5594 is the bulk of it

this allows passing an HTTP error code to use, I think is the main difference.

File: httperror/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+func New() *App {
+    r := echo.New()
+
+    // This is the important bit.
+    r.HTTPErrorHandler = httperror.HTTPErrorHandler
+
+    return &App{
+        router: echo.New(),
```
Comment (slr71) [inline httperror/README.md]: I think this should refer to the router that was created above:

```suggestion
        router: r,
```

**Review by slr71 — APPROVED** [review]
Looks good! I just had one minor suggestion for the README.

**Comment by ianmcorvidae** [comment]
Looks good to me!

**Pull request merged**